backend/ethereum/wallet/hd/wallet_test.go

```diff
@@ -122,19 +120,15 @@ func newSetup(t require.TestingT, prng *rand.Rand) (*test.Setup, accounts.Wallet
 	require.NotNil(t, acc)
 
 	addressNotInWallet := ethwallettest.NewRandomAddress(prng)
-	var buff bytes.Buffer
-	err = perunio.Encode(&buff, &addressNotInWallet)
-	if err != nil {
-		panic(err)
-	}
-	addrEncoded := buff.Bytes()
+	addrMarshalled, err := addressNotInWallet.MarshalBinary()
+	require.NoError(t, err)
 
 	return &test.Setup{
-		Wallet:          hdWallet,
-		AddressInWallet: acc.Address(),
-		Backend:         new(ethwallet.Backend),
-		AddressEncoded:  addrEncoded,
-		ZeroAddress:     ethwallet.AsWalletAddr(common.Address{}),
-		DataToSign:      dataToSign,
+		Wallet:            hdWallet,
+		AddressInWallet:   acc.Address(),
+		Backend:           new(ethwallet.Backend),
+		AddressMarshalled: addrMarshalled,
```

Comment:
Would be interested to know why do you think so ?

I did a rename here, because `Address` type has both `Encode` and `MarshalBinary` methods defined on it. And we have switched from using `Encode` to `MarshalBinary`.

Reply:
The [`Address` type](https://github.com/hyperledger-labs/go-perun/blob/5af1ef7b74f60e3d4c9f946ebdd80da10c850300/wallet/address.go#L31) does not have an `Encode` function anymore.